engine/src/main/java/org/terasology/rendering/nui/widgets/UIText.java

```diff
@@ -82,6 +82,9 @@
     @LayoutConfig
     protected boolean readOnly;
 
+    @LayoutConfig
+    private boolean passwordMode;
```

Comment:
Many UI frameworks have a separate class (derived from the default text control). Have you thought about creating a UIPasswordText instead? It could be too much duplication, though ..